lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -65,7 +65,12 @@ function dosomething_rogue_send_reportback_to_rogue($values, $user = NULL, $fail
   // aware of a user's northstar id. If it doesn't find one, we just grab it
   // from northstar directly.
   if (!$northstar_id) {
-    $northstar_user = dosomething_northstar_get_user($user->uid, 'drupal_id');
+    if ($user->uid) {
+      $northstar_user = dosomething_northstar_get_user($user->uid, 'drupal_id');
+    } else {
+      $northstar_user = dosomething_northstar_get_user($values['northstar_id']);
+      $user = user_load($northstar_user->drupal_id);
```

Comment:
that's where we're running into the issue! So some of the records in the `dosomething_rogue_failed_task_log` do not have a `drupal_id`, so `$user` in the line you linked to is `NULL`.